pynicotine/gtkgui/dialogs/preferences.py

```diff
@@ -671,37 +656,62 @@ def on_add_shared_folder(self, *_args):
     def on_edit_shared_folder_response(self, dialog, _response_id, iterator):
 
         new_virtual_name = dialog.get_entry_value()
-        new_is_buddy_only = dialog.get_option_value()
+        new_available_to = dialog.get_second_entry_value()
 
-        self._edit_shared_folder(iterator, new_virtual_name, new_is_buddy_only)
+        virtual_name = self.shares_list_view.get_row_value(iterator, "virtual_name")
+        available_to = self.shares_list_view.get_row_value(iterator, "available_to")
+
+        if new_virtual_name == virtual_name and new_available_to == available_to:
+            return
+
+        self.rescan_required = True
+
+        folder_path = self.shares_list_view.get_row_value(iterator, "folder")
+        group_names = {
+            _("Everyone"): "public",
+            _("Buddies"): "buddy",
+            _("Trusted buddies"): "trusted"
```

Comment:
Consider creating a constant in the shares module for the share group names.

Reply:
These are quite UI-specific, so probably won't do this.